examples/restore_login.py

```diff
@@ -1,71 +1,90 @@
+#!/usr/bin/env python3
+
 import asyncio
 import json
 import os
 import sys
+import getpass
 
 from nio import AsyncClient, LoginResponse
 
 CONFIG_FILE = "restore_login_config.json"
 
 # Check out main() below to see how it's done.
 
-def write_details_to_disk(resp: LoginResponse) -> None:
+
+def write_details_to_disk(resp: LoginResponse, home_server) -> None:
     """Writes the required login details to disk so we can log in later without
     using a password.
 
     Arguments:
         resp {LoginResponse} -- the successful client login response.
+        home_server -- the name of the homeserver, e.g. "https://matrix.example.org"
     """
     # open the config file in write-mode
     with open(CONFIG_FILE, "w") as f:
         # write the login details to disk
         json.dump(
             {
-            "access_token": resp.access_token,
-            "device_id": resp.device_id,
-            "user_id": resp.user_id
+                "home_server": home_server,  # e.g. "https://matrix.example.org"
+                "access_token": resp.access_token,  # long cryptographic access token
+                "device_id": resp.device_id,  # device ID, 10 uppercase letters
+                "user_id": resp.user_id  # e.g. "@user:example.org"
             },
             f
         )
 
 
 async def main() -> None:
-    client = AsyncClient("https://matrix.example.org", "@alice:example.org")
 
     # If there are no previously-saved credentials, we'll use the password
     if not os.path.exists(CONFIG_FILE):
-        resp = await client.login("hunter2")
+        print("First time use. Did not find credential file. Asking for homeserver, user, and password to create credential file.")
+        home_server = "https://matrix.example.org"
```

Comment:
How come there are default values for `home_server` and `user_id`? You immediately require user input so the default value won't ever be used, it'll be overwritten.

If you want to suggest the type, you can also use `homeserver: str = input(...)`

Reply:
There are 2 reasons why I have done that:
- a casual reader that just browses thru Github or through the code gets a better understanding what kind of input is expected
- someone who really is executing this little program sees in the prompt an example value which will be helpful for newcomers, from the example they will know what value to enter at the prompt.

input() always returns a string. I don't think it is necessary to underline that this is a string. The examples speak for themselves. 

I left that part unchanged in commit 2.

Reply:
Sure, that's fine